backend/main/test_utils/strategy_utils.go

```diff
@@ -84,3 +114,88 @@ func (otu *OverflowTestUtils) GenerateListOfVotes(proposalId int, count int) *[]
 
 	return &votes
 }
+
+func (otu *OverflowTestUtils) GenerateListOfVotesWithNFTs(
+	proposalId int,
+	count int,
+	contract *shared.Contract,
+) (*[]VoteWithBalance, error) {
+	var votes []VoteWithBalance
+	choices := []string{"a", "b"}
+	for i := 0; i < count; i++ {
+		accountName := "user" + strconv.Itoa(i+1)
+		otu.SetupAccountForNFTs(accountName)
+		otu.MintNFT("user1", accountName)
+
+		addr := otu.ResolveUser(i + 1)
+		randomNumber := rand.Intn(2)
+		choice := choices[randomNumber]
+		v := models.Vote{
+			Proposal_id: proposalId, Addr: addr, Choice: choice,
+		}
+
+		nftIds, err := otu.Adapter.GetNFTIds(addr, contract)
+		if err != nil {
+			return nil, err
+		}
+
+		vote := otu.CreateNFTVote(v, nftIds, contract)
+
+		balance := 100000000 * (i + 1)
+		vote.Primary_account_balance = uint64(balance)
+		vote.Staking_balance = uint64(balance * 5)
+		vote.Block_height = uint64(0)
+
+		votes = append(votes, vote)
+	}
+
+	return &votes, nil
+}
+
+func (otu *OverflowTestUtils) CreateNFTVote(v models.Vote, ids []interface{}, contract *shared.Contract) VoteWithBalance {
```

Comment:
any reason ids is type `[]interface{}` rather than `[]string` or `[]uint`?

Reply:
its the type that returns from the function `CadenceValueToInterface` which is used when parsing the returned values from the cadence script
...
case cadence.Dictionary:
   result := map[string]interface{}{}
```